bin/create-hierarchy

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+import argparse
+import pandas as pd
+from pathlib import Path
+
+base = Path(__file__).resolve().parent.parent
+
+import sys
+sys.path.insert(0, str(base / "lib"))
+from utils.hierarchy_dataframe import hierarchy_dataframe
+
+
+HIERARCHY_COLUMNS = ['region', 'country', 'division']
+
+parser = argparse.ArgumentParser(
+    description="Create a known location hierarchy from given metadata that "
+        "can be used to validate future metadata and annotations. The output "
+        "hierarchy TSV is printed to stdout, and you likely want to redirect "
+        "it to a new file.",
+    formatter_class=argparse.RawTextHelpFormatter
+)
+
+parser.add_argument(
+    "metadata",
+    nargs="?",
+    default="s3://nextstrain-ncov-private/metadata.tsv.gz",
```

Comment:
This path will want to be updated during/after the rebase.

Reply:
Thank Tom for your feedback. I'll try to address it in comments, but I'm happy to jump on a call with you to further discuss the intended functionality here.

> For example, why does `ingest` run `create-hierarchy` to produce a local version of `location_hierarchy.tsv` from the _existing_ `s3://nextstrain-ncov-private/metadata.tsv.gz` but `transform` reads from `s3://nextstrain-ncov-private/location_hierarchy.tsv.gz`? And then the local version of `location_hierarchy.tsv` is uploaded to S3. It seems like at no point is an updated `location_hierarchy.tsv` made from the _new_ metadata.

You're right, this seems like an oversight on my end. I believe this branch got large enough that I started to lose sight of the bigger picture here. It seems like we should create a new location hierarchy at the end of ingest using new metadata and upload it to S3.

> I'm also confused about how `draft_metadata.tsv` is intended to be used. It is written but never uploaded anywhere…? (When we talked about a draft metadata table vs. the final metadata table before, the idea was that the the draft metadata makes it easier to compare the effect of annotations because the draft metadata + annotations = final metadata.)

You're right, it's not used anywhere here, just written. I thought it would be useful for nCoV build maintainers but wasn't entirely clear on how it could be useful other than through its pure existence. Do you have any recommendations on how to make it more useful? 

> It also strikes me as a bit strange to bundle up production of the `new_location_hierarchies.tsv` into `transform` rather than as a separate diff program that takes the old and new metadata files. I'd think an old vs. new diff is what we want, but maybe I'm missing the point of comparing against `location_hierarchy.tsv` instead?

The intent is to capture any new resolution combinations. Are you thinking an old vs new diff of unique rows of the resolution (e.g. region, country, division) columns? 

> If you could rebase this onto the latest version of master, that'd also be appreciated.

I will certainly do that.